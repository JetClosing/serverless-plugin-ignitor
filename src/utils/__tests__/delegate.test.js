const { create, events } = require('../delegate');

const rates = {
  'rate(3 minutes)': [
    {
      lambda: 'Example-dev-hello',
      input: {
        ignitor: true,
      },
    },
  ],
};

test('creates delegate', () => {
  expect(create(rates)).toMatchSnapshot();
});


test('create schedules', () => {
  expect(events(rates)).toMatchSnapshot();
});
