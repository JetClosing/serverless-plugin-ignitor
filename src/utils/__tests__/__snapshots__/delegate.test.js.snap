// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create schedules 1`] = `
Array [
  Object {
    "schedule": Object {
      "enabled": true,
      "input": Object {
        "rate": "rate(3 minutes)",
      },
      "rate": "rate(3 minutes)",
    },
  },
]
`;

exports[`creates delegate 1`] = `
"const AWS = require('aws-sdk');

const L = new AWS.Lambda({ apiVersion: '2015-03-31' });

const map = {
  \\"rate(3 minutes)\\": [
    {
      \\"lambda\\": \\"Example-dev-hello\\",
      \\"input\\": {
        \\"ignitor\\": true
      }
    }
  ]
};

const delegate = (evt, ctx, callback) => {
  console.log('Event', JSON.stringify(evt, null, 2));
  if (!evt.rate) {
    throw new Error('No rate found to delegate');
  }

  const listeners = (map[evt.rate] || []);
  console.log('listeners', JSON.stringify(listeners, null, 2));
  Promise.all(
    listeners
      .map(({ lambda, input }) => L.invoke({
        FunctionName: lambda,
        InvocationType: 'Event',
        Payload: JSON.stringify(input),
      }).promise())
  ).then((res) => {
    callback(null, { success: true });
  })
  .catch((err) => {
    callback(err.message);
  });
};

module.exports = {
  handler: delegate,
};"
`;
